### PR TITLE
Log more response errors

### PR DESCRIPTION
**Pull request by Zelldon**
Previously we only logged (not sure but current assumption) BAD_REQUEST returned from SAAS

Related to https://github.com/zeebe-io/zeebe-cluster-testbench/issues/889

### CODE REVIEW

File: cloud-client/src/main/java/io/zeebe/clustertestbench/cloud/filter/BadResponseFilter.java [hunk 1]
```diff
@@ -7,26 +7,34 @@
...
+  private static final Logger LOG = LoggerFactory.getLogger(BadResponseFilter.class);
 
   @Override
   public void filter(
       final ClientRequestContext requestContext, final ClientResponseContext responseContext)
       throws IOException {
+    if (responseContext.getStatus() > 399) {
 
-    if (responseContext.getStatus() == 400) {
       final String requestBody = new ObjectMapper().writeValueAsString(requestContext.getEntity());
```
Comment (megglos) [inline cloud-client/src/main/java/io/zeebe/clustertestbench/cloud/filter/BadResponseFilter.java:23]: 🔧  pre-existing and minor given the context but while we are here anyway, could just make the objectmapper a static field to initialize once and reuse as it's relatively expensive to construct

File: core/src/test/java/io/zeebe/clustertestbench/handler/FailedServiceShouldFailResponseTest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    client = ClientBuilder.newBuilder().register(BadResponseFilter.class).build();
+    final var target = (ResteasyWebTarget) client.target("http://localhost:8081");
+    final var cloudAPIClient = target.proxy(InternalCloudAPIClient.class);
+
+    spyInternalApiClient = spy(cloudAPIClient);
+    localhost = HttpServer.create(new InetSocketAddress("localhost", 8081), 1);
+    localhost.start();
+    activatedJobStub = jobClientStub.createActivatedJob();
+    sutCreateGenerationHandler = new CreateGenerationInCamundaCloudHandler(spyInternalApiClient);
+  }
```
Comment (megglos) [inline core/src/test/java/io/zeebe/clustertestbench/handler/FailedServiceShouldFailResponseTest.java:46]: 💭 I wonder if we could make the test setup simple, maybe like resteasy does response filter testing here https://github.com/resteasy/resteasy/blob/main/testsuite/unit-tests/src/test/java/org/jboss/resteasy/test/client/ClientResponseFilterTest.java but also fine to keep it that way

**Review by megglos — APPROVED** [review]
LGTM! Just some ideas/suggestions.

File: core/src/test/java/io/zeebe/clustertestbench/handler/FailedServiceShouldFailResponseTest.java (hunk 2 above)
Comment (Zelldon) [reply core/src/test/java/io/zeebe/clustertestbench/handler/FailedServiceShouldFailResponseTest.java:46]: Actually, I wrote the test, due to lack of knowledge and I wanted to verify how the client behaves when there is a service but doesn't answer or returns an error. This was not clear nor transparent to me. But I guess we can also remove it again.

**Pull request merged**